common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/TokenResponse.java

```diff
@@ -7,4 +7,166 @@
  * support an extension to allow the authorization server / openid provider to send back additional information
  */
 public class TokenResponse {
+
+    /**
+     * RECOMMENDED.  The lifetime in seconds of the access token.  For
+     * example, the value "3600" denotes that the access token will
+     * expire in one hour from the time the response was generated.
+     * If omitted, the authorization server SHOULD provide the
+     * expiration time via other means or document the default value.
+     *
+     * @see <a href="https://tools.ietf.org/html/rfc6749#section-5.1">RFC 6749 - Successful Response</a>
+     */
+    protected int mExpiresIn;
```

Comment:
Per [this example](https://tools.ietf.org/html/rfc6749#section-5.1) in the RFC-6748, `expires_in` is returned as a JSON number type.

Per the example in [this documentation for AAD](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-protocols-oauth-code) the `expires_in` field is shown as a `String`.

Reply:
Should figure out which it really is, or consider using [`JSONObject#getString(String)`](https://developer.android.com/reference/org/json/JSONObject.html#getString(java.lang.String)) to parse/coerce